### DISCUSSION TITLE
DB cache leak or misconfiguration?

### DISCUSSION THREAD

**Discussion by Franck31** [netdata/netdata#17168] [Q&A]
Hello Netdata Team,

I hope this message finds you well. I've encountered significant issues with my Netdata installation, particularly concerning out-of-memory (OOM) errors and parent process failures. Upon closer inspection, I've identified a memory leak directly linked to the main cache of the db-engine, which continues to grow unchecked until the server becomes unstable and crashes. This pattern repeats consistently.

Upon reviewing the provided dashboards, it's evident that the issue is closely tied to dbengine_main_cache_memory->HOT.

I'm reaching out to inquire if there are any known solutions to mitigate this problem. Perhaps there are configuration options I may have overlooked or misconfigured.

For context, I'm using Netdata version 1.40.1 on Amazon EKS (Elastic Kubernetes Service).

Attached to this message, you'll find screenshots illustrating the database leak and my current configuration settings.

showing the leak on the cache: 
![image](https://github.com/netdata/netdata/assets/13457574/63d98eeb-3ef3-4d80-b784-9a1ef5874aec)

more on detail: 
![image](https://github.com/netdata/netdata/assets/13457574/60ded0d3-85ed-42e0-9b65-1c8b1680cf5e)

My netdata parent CFG: 
![image](https://github.com/netdata/netdata/assets/13457574/498d341d-adb7-4976-8f16-552df9d4587b)

and my db-engine stats: 
![image](https://github.com/netdata/netdata/assets/13457574/02521f13-26a1-488d-b577-e803dd89038c)

**Comment by tkatsoulas** [comment]
Hello @Franck31, this a pretty old version. Please upgrade to a new one and see if the problem persists

**Reply by Franck31** [reply]
I've encountered the same issue with Netdata versions 1.44.2 and 1.44.1. Despite upgrading, I'm still experiencing the same pattern of out-of-memory errors and parent process failures associated with the main cache of the db-engine. 

So I think it might be a misconfiguration or a persistent leak problem causing this issue.
